### PR TITLE
Feature/sep2

### PR DESCRIPTION
**Pull request by rob-calvert**
SEP2 driver and agent.

Please note that this pull request has a dependency on Andrew's pull request #1388. Specifically, it depends on WebAgent in volttron/platform/web.py. (That pull request has now been merged.)

---
This change is [Reviewable](https://reviewable.io/reviews/volttron/volttron/1406)

### CODE REVIEW

**Comment by kmonson** [comment]
The failed tests are the platform timeout cruft we haven't sorted out yet. I'll review this again and get back to you.

File: services/core/SEP2Agent/sep2/agent.py [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+        self.load_shed_device_category = load_shed_device_category
+        self.timezone = timezone
+
+        self.default_config = {"agentid": agentid,
+                               "device_config": device_config,
+                               "sep2_server_sfdi": sep2_server_sfdi,
+                               "sep2_server_lfdi": sep2_server_lfdi,
+                               "load_shed_device_category": load_shed_device_category,
+                               "timezone": timezone}
+
```
Comment (rob-calvert) [inline services/core/SEP2Agent/sep2/agent.py]: Thanks for the recommendations, Kyle, they are much appreciated. I've implemented all of the other changes you suggested here, and I've added self.vip.config.set_default("config", self.default_config), but I wasn't able to eliminate the redundant calls to register_endpoints() and register_devices(). If I do that, the pytests fail.

File: services/core/MasterDriverAgent/master_driver/interfaces/sep2.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            volttron-ctl config list platform.driver
+    """
+
+    def __init__(self, **kwargs):
+        super(Interface, self).__init__(**kwargs)
+        self.sfdi = ''
+        self.sep2_agent_id = DEFAULT_SEP2_AGENT_ID
+        self.cache_expiration_secs = DEFAULT_CACHE_EXPIRATION_SECS
+
+    def configure(self, config_dict, registry_config_str):
```
Comment (rob-calvert) [inline services/core/MasterDriverAgent/master_driver/interfaces/sep2.py]: Done

File: services/core/SEP2Agent/sep2/agent.py [hunk 3]
```diff
@@ -0,0 +1,570 @@
...
+    """
+    try:
+        config = utils.load_config(config_path)
+    except StandardError:
+        config = {}
+
+    if not config:
+        _log.info("Using SEP2 Agent defaults for starting configuration.")
+
+    agentid = config.get('agentid', 'SEP2-agent')
```
Comment (rob-calvert) [inline services/core/SEP2Agent/sep2/agent.py]: Fixed

File: services/core/SEP2Agent/sep2/agent.py [hunk 4]
```diff
@@ -0,0 +1,570 @@
...
+        SEP2 data is exposed via get_point(), get_points() and set_point() calls.
+        A SEP2 device driver (sep2.py under MasterDriverAgent) can be configured,
+        which gets and sets data by sending RPCs to this agent.
+
+        For further information about this subsystem, please see the VOLTTRON
+        SEP 2.0 DER Support specification, which is located in VOLTTRON readthedocs
+        under specifications/sep2_agent.html.
+
+        This agent can be installed as follows:
+            export VIP_SOCKET="ipc://$VOLTTRON_HOME/run/vip.socket"
```
Comment (rob-calvert) [inline services/core/SEP2Agent/sep2/agent.py]: Thanks for pointing it out, I removed the comment.

File: services/core/SEP2Agent/sep2/agent.py [hunk 5]
```diff
@@ -0,0 +1,570 @@
...
+            raise SEP2Exception("{0} not a valid point name.".format(point_name))
+
+    @RPC.export
+    def get_points(self, sfdi):
+        end_device = self.get_end_device(sfdi=sfdi)
+        try:
+            end_device_points = {}
+            for point in sep2.ALL_POINTS:
+                end_device_points[point] = getattr(end_device, point)
+            return json.dumps(end_device_points)
```
Comment (rob-calvert) [inline services/core/SEP2Agent/sep2/agent.py]: Oops, thanks for catching it. Fixed.

**Pull request merged**